pubsub/google/cloud/pubsub_v1/publisher/futures.py

```diff
@@ -14,156 +14,27 @@
-        err = self.exception(timeout=timeout)
-        if err is None:
-            return self._result
-        raise err
-
-    def exception(self, timeout=None, _wait=1):
-        """Return the exception raised by the call, if any.
-
-        This blocks until the message has successfully been published, and
-        returns the exception. If the call succeeded, return None.
-
-        Args:
-            timeout (Union[int, float]): The number of seconds before this call
-                times out and raises TimeoutError.
-
-        Raises:
-            TimeoutError: If the request times out.
-
-        Returns:
-            Exception: The exception raised by the call, if any.
-        """
-        # Wait until the future is done.
-        if not self._completed.wait(timeout=timeout):
-            raise exceptions.TimeoutError('Timed out waiting for result.')
-
-        # If the batch completed successfully, this should return None.
-        if self._result is not None:
-            return None
-
-        # Okay, this batch had an error; this should return it.
-        return self._exception
-
-    def add_done_callback(self, fn):
-        """Attach the provided callable to the future.
-
-        The provided function is called, with this future as its only argument,
-        when the future finishes running.
-        """
-        if self.done():
-            fn(self)
-        self._callbacks.append(fn)
-
-    def set_result(self, result):
-        """Set the result of the future to the provided result.
-
-        Args:
-            result (str): The message ID.
-        """
-        # Sanity check: A future can only complete once.
-        if self._result is not None or self._exception is not None:
-            raise RuntimeError('set_result can only be called once.')
-
-        # Set the result and trigger the future.
-        self._result = result
-        self._trigger()
-
-    def set_exception(self, exception):
-        """Set the result of the future to the given exception.
-
-        Args:
-            exception (:exc:`Exception`): The exception raised.
-        """
-        # Sanity check: A future can only complete once.
-        if self._result is not None or self._exception is not None:
-            raise RuntimeError('set_exception can only be called once.')
-
-        # Set the exception and trigger the future.
-        self._exception = exception
-        self._trigger()
-
-    def _trigger(self):
-        """Trigger all callbacks registered to this Future.
-
-        This method is called internally by the batch once the batch
-        completes.
-
-        Args:
-            message_id (str): The message ID, as a string.
-        """
-        self._completed.set()
-        for callback in self._callbacks:
-            callback(self)
+    # The publishing-side subclass does not need any special behavior
+    # at this time.
+    #
+    # However, there is still a subclass so that if someone attempts
```

Comment:
I don't think we should bother with this distinction - in fact, I don't think we should bother with the distinction that these are anything other that `google.api_core.future.Future` instances.